static/app/components/dropdownMenuItemV2.tsx

```diff
@@ -105,137 +111,160 @@ type Props = {
+        onClose,
+        closeOnSelect,
+        isSubmenuTrigger = false,
+        renderAs = 'li' as React.ElementType,
+        router,
+        ...submenuTriggerProps
+      },
+      submenuTriggerRef
+    ) => {
+      const [isHovering, setIsHovering] = useState(false);
+      const ref = (submenuTriggerRef ?? useRef(null)) as React.RefObject<HTMLLIElement>;
+      const isDisabled = state.disabledKeys.has(node.key);
+      const isFocused = state.selectionManager.focusedKey === node.key;
+      const item = node.value;
 
-    // Open submenu on hover
-    const {hoverProps} = useHover({onHoverChange: setIsHovering});
-    useEffect(() => {
-      if (isHovering && isFocused) {
+      const actionHandler = () => {
         if (isSubmenuTrigger) {
           state.selectionManager.select(node.key);
-        } else {
-          state.selectionManager.clearSelection();
+          return;
         }
-      }
-    }, [isHovering, isFocused]);
+        item.onAction?.(item.key);
+        item.to && router.push(item.to);
+      };
 
-    // Open submenu on arrow right key press
-    const {keyboardProps} = useKeyboard({
-      onKeyDown: e => {
-        if (isSubmenuTrigger && e.key === 'ArrowRight') {
-          state.selectionManager.select(node.key);
-        } else {
-          e.continuePropagation();
+      // Open submenu on hover
+      const {hoverProps} = useHover({onHoverChange: setIsHovering});
+      useEffect(() => {
+        if (isHovering && isFocused) {
+          if (isSubmenuTrigger) {
+            state.selectionManager.select(node.key);
+          } else {
+            state.selectionManager.clearSelection();
+          }
         }
-      },
-    });
+      }, [isHovering, isFocused]);
 
-    // Manage interactive events & create aria- attributes
-    const {menuItemProps, labelProps, descriptionProps} = useMenuItem(
-      {
-        key: node.key,
-        onAction: actionHandler,
-        onClose,
-        closeOnSelect,
-        isDisabled,
-      },
-      state,
-      ref
-    );
+      // Open submenu on arrow right key press
+      const {keyboardProps} = useKeyboard({
+        onKeyDown: e => {
+          if (isSubmenuTrigger && e.key === 'ArrowRight') {
+            state.selectionManager.select(node.key);
+          } else {
+            e.continuePropagation();
+          }
+        },
+      });
 
-    // Merged menu item props, class names are combined, event handlers chained,
-    // etc. See: https://react-spectrum.adobe.com/react-aria/mergeProps.html
-    const props = mergeProps(
-      submenuTriggerProps,
-      menuItemProps,
-      hoverProps,
-      keyboardProps
-    );
-    const {
-      details,
-      leadingItems,
-      leadingItemsSpanFullHeight,
-      trailingItems,
-      trailingItemsSpanFullHeight,
-    } = item;
-    const label = node.rendered ?? item.label;
-    const showDividers = item.showDividers && !isLastNode;
+      // Manage interactive events & create aria- attributes
```

Comment:
```suggestion
      // Manage interactive events & create aria-attributes
```

Reply:
should this be together?

Reply:
think it should be `aria attributes` actually